Add CloseCommand

jakq opened the pull request:
Closes #56 

Add a boolean variable isOpen to check if the eatery is open or closed.
Add CloseCommand.
Add CloseCommandParser.
Add the relevant Test classes.

jakq committed:
Refactor Person to Eatery

v1.0 with 3 EditCommandTest failures.

jakq committed:
Merge remote-tracking branch 'upstream/master' into refactor-person-to-eatery

jakq committed:
Removed 2 test cases from EditCommandTest

jakq committed:
Update BetterModelClassDiagram.puml

jakq committed:
Remove Email and Phone class

jakq committed:
Remove Email Phone multiplicity

jakq committed:
Refactor Person to Eatery edit1

jakq committed:
Merge branch 'refactor-person-to-eatery' of https://github.com/jakq/main into refactor-person-to-eatery

jakq committed:
Add Close Command

jakq committed:
Restore whiteline to AddressBookTest

jakq committed:
Merge remote-tracking branch 'upstream/master' into close-open-eatery

jakq committed:
Merge remote-tracking branch 'upstream/master' into close-open-eatery

jakq committed:
Merge remote-tracking branch 'upstream/master' into close-open-eatery

# Conflicts:
#	src/main/java/seedu/address/logic/Logic.java
#	src/main/java/seedu/address/logic/LogicManager.java
#	src/main/java/seedu/address/logic/commands/EditCommand.java
#	src/main/java/seedu/address/logic/parser/AddCommandParser.java
#	src/main/java/seedu/address/logic/parser/ParserUtil.java
#	src/main/java/seedu/address/model/AddressBook.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/model/eatery/Eatery.java
#	src/main/java/seedu/address/model/eatery/UniqueEateryList.java
#	src/main/java/seedu/address/model/util/SampleDataUtil.java
#	src/main/java/seedu/address/storage/JsonAdaptedEatery.java
#	src/main/java/seedu/address/storage/JsonSerializableAddressBook.java
#	src/main/java/seedu/address/ui/EateryCard.java
#	src/main/java/seedu/address/ui/MainWindow.java
#	src/main/resources/view/EateryListCard.fxml
#	src/main/resources/view/EateryListPanel.fxml
#	src/main/resources/view/MainWindow.fxml
#	src/test/data/JsonSerializableAddressBookTest/duplicateEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/invalidEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/typicalEateriesAddressBook.json
#	src/test/java/seedu/address/logic/LogicManagerTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/ClearCommandTest.java
#	src/test/java/seedu/address/logic/commands/DeleteCommandTest.java
#	src/test/java/seedu/address/logic/commands/EditCommandTest.java
#	src/test/java/seedu/address/logic/commands/FindCommandTest.java
#	src/test/java/seedu/address/logic/commands/ListCommandTest.java
#	src/test/java/seedu/address/logic/parser/AddCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/AddressBookParserTest.java
#	src/test/java/seedu/address/logic/parser/EditCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/ParserUtilTest.java
#	src/test/java/seedu/address/model/AddressBookTest.java
#	src/test/java/seedu/address/model/ModelManagerTest.java
#	src/test/java/seedu/address/model/eatery/EateryTest.java
#	src/test/java/seedu/address/model/eatery/UniqueEateryListTest.java
#	src/test/java/seedu/address/storage/JsonAdaptedEateryTest.java
#	src/test/java/seedu/address/testutil/EateryBuilder.java
#	src/test/java/seedu/address/testutil/EateryUtil.java
#	src/test/java/seedu/address/testutil/EditEateryDescriptorBuilder.java
#	src/test/java/seedu/address/testutil/TypicalEateries.java

wpuffs commented on src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:
```diff
@@ -57,7 +57,7 @@ public void execute_validIndexFilteredList_success() {
         String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_EATERY_SUCCESS, eateryToDelete);
 
-        Model expectedModel = new ModelManager(model.getAddressBook(), model.getFeedList(), new UserPrefs());
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), model.getFeedList(), new UserPrefs());
```
why you use modelmanager instead of model ah isn't model the interface class

jakq replied:
idk the previous test case, like the one above uses modelmanager for expected model, so i just changed it ahhh

wpuffs replied:
ah okie

wpuffs commented on src/main/java/seedu/address/logic/commands/CloseCommand.java:
```diff
@@ -0,0 +1,86 @@
+     * Creates and returns a {@code Eatery} with the details of {@code eateryToEdit}
+     * edited with {@code editEateryDescriptor}.
+     */
+    private static Eatery createClosedEatery(Eatery eateryToClose) {
+        assert eateryToClose != null;
+
+        Name name = eateryToClose.getName();
+        Address address = eateryToClose.getAddress();
+        Category category = eateryToClose.getCategory();
+        Set<Tag> tags = eateryToClose.getTags();
+
+        return new Eatery(name, false, address, category, tags);
+    }
```
just curious de why do you create a new eatery instead of just editing the current eatery? also, do you get rid of this current eatery when u close the eatery? then if u reopen the eatery do u create a new eatery with open or do u just reuse the old eatery which was already open?

jakq replied:
oh so, i follow editcommand style, where they just replace the old one with the new one

wpuffs replied:
ah so then if u reopen the eatery then u will follow the same logic and create a totally new eatery with the isOpen set to true?

wpuffs commented on src/main/java/seedu/address/model/eatery/Eatery.java:
```diff
@@ -15,26 +15,46 @@
      */
     public Eatery(Name name, Address address, Category category, Set<Tag> tags) {
-        requireAllNonNull(name, address, category, tags);
+        requireAllNonNull(name, address, tags);
```
why did you delete category?

jakq replied:
oops my bad, might have accidentally deleted that. But the user cant create an eatery without category at the start?

wpuffs replied:
yeah they can't hahaha like if u refer to our userguide, only tags are optional? so i think u can remove requireAllNonNull for tags? idk

wpuffs commented on src/main/java/seedu/address/ui/EateryCard.java:
```diff
@@ -44,8 +44,8 @@ public EateryCard(Eatery eatery, int displayedIndex) {
         name.setText(eatery.getName().fullName);
-        category.setText(eatery.getCategory().getName());
         address.setText(eatery.getAddress().value);
+        category.setText(eatery.getCategory().getName());
```
why u move

jakq replied:
move as in down? I followed the convention of category after address

wpuffs commented on src/main/resources/view/EateryListPanel.fxml:
```diff
@@ -2,4 +2,4 @@
 <?import javafx.scene.control.ListView?>
 
-<ListView xmlns:fx="http://javafx.com/fxml/1" fx:id="eateryListView" styleClass="plain-border-box"/>
+<ListView xmlns:fx="http://javafx.com/fxml/1" fx:id="eateryListView" styleClass="plain-border-box"/>
```
this is the one causing ur travis to fail btw

wpuffs commented on src/test/data/JsonSerializableAddressBookTest/duplicateEateryAddressBook.json:
```diff
@@ -2,11 +2,9 @@
     "name": "Alice Pauline",
-    "address": "4th street",
-    "category": "Western"
+    "address": "4th street"
```
why u change this ah lmao soz

jakq replied:
oh tbh idk, but i dont think duplicate eatery is based on category anyway?

wpuffs commented on src/test/java/seedu/address/testutil/EateryBuilder.java:
```diff
@@ -74,7 +89,10 @@ public EateryBuilder withCategory(String category) {
 
+    public Eatery buildClose() {
+        return new Eatery(name, false, address, category, tags);
+    }
```
why do you have a buildClose() when you can like just use .withIsOpen(false).build()?

jakq replied:
haha omy yea i created too many methods LOL

wpuffs replied:
does this mean u gna delete buildClose() HAHAHA i think it's more consistent if u do .withIsOpen()..

wpuffs commented on src/test/java/seedu/address/storage/JsonSerializableAddressBookTest.java:
```diff
@@ -23,8 +23,8 @@ public void toModelType_typicalEateriesFile_success() throws Exception {
         AddressBook addressBookFromFile = dataFromFile.toModelType();
-        AddressBook typicalEaterysAddressBook = TypicalEateries.getTypicalAddressBook();
-        assertEquals(addressBookFromFile, typicalEaterysAddressBook);
+        AddressBook typicalEateriesAddressBook = TypicalEateries.getTypicalAddressBook();
```
why u do this what did u even change HAHAHA

jakq replied:
typicalEaterysAddressBook to typicalEateriesAddressBook

wpuffs replied:
oh lolsies

wpuffs commented on src/test/java/seedu/address/logic/commands/CloseCommandTest.java:
```diff
@@ -0,0 +1,103 @@
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Eatery closedEatery = new EateryBuilder().withTags("friends").buildClose();
```
same comment as the in your eaterybuilder, why buildClose() instead of .withIsOpen(false).build()?

jakq committed:
Fix CloseCommand

1. Remove buildClose() from EateryBuilder
2. Add EOF line to EateryListPanel.fxml
3. Fix duplicateEateryAddressBook.json
4. Require nonnull Category for Eatery

wpuffs commented on src/main/java/seedu/address/model/eatery/Eatery.java:
```diff
@@ -26,7 +26,7 @@
      */
     public Eatery(Name name, Address address, Category category, Set<Tag> tags) {
-        requireAllNonNull(name, address, tags);
+        requireAllNonNull(name, address, category, tags);
```
i think can remove tags!!! bc tags is optional hahaha right?? right idk

wpuffs reviewed (changes requested):
good job bboy

jakq committed:
Remove tags from requireAllNonNull in Eatery
